### PR TITLE
Csv upload v2

### PR DESCRIPTION
**Pull request by cchyung**
second pass at CSV upload integrated with new pop up menu

### CODE REVIEW

File: src/components/dashboard/menu-table/popup-forms.js [hunk 1]
```diff
@@ -433,11 +435,140 @@ const QRCodeForm = (props) => {
...
+            const reader = new FileReader()      
+            reader.onabort = () => console.error('file reading was aborted')
+            reader.onerror = () => {
+                console.error('file reading has failed')
+                setErrorMessage("Error uploading file")
+            }
+            reader.onload = () => {
+            // Do whatever you want with the file contents
+              const fileContent = reader.result
+              setContent(fileContent)
```
Comment (Tony1023) [inline src/components/dashboard/menu-table/popup-forms.js:491]: I happened to discover that if you uploaded an empty file the view doesn't change or say anything that reminds the user an empty file is uploaded...

File: src/pages/dashboard/menu.js [hunk 2]
```diff
@@ -82,10 +63,9 @@ const MenuPage = () => {
                 <Column>
                     <MenuContainer>
                         <MenuSelector updateMenuSelection={updateMenuSelection} menuId={menuId} />
```
Comment (Tony1023) [inline src/pages/dashboard/menu.js:65]: Is the selector not styled yet?

File: src/components/dashboard/menu-table/popup-forms.js [hunk 3]
```diff
@@ -433,11 +435,140 @@ const QRCodeForm = (props) => {
...
 }
 
+const StyledUploadCSVModal = styled.div`
+    input[type="checkbox"] {
+        display: inline-block;
+    }
+
+    input[type="file"] {
+
+    }
```
Comment (Tony1023) [inline src/components/dashboard/menu-table/popup-forms.js:445]: Remove?

File: src/components/dashboard/menu-table/menu-table.js [hunk 4]
```diff
@@ -201,38 +193,6 @@ const MenuTable = (props) => {
...
-      for (var i = 1; i < allTextLines.length; i++) {
-          var data = allTextLines[i].split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
-          if (data.length == headers.length) {
-              var tarr = []
-              for (var j = 0; j < headers.length; j++) {
-                  tarr.push(data[j].replace(/['"]+/g, ''));
-              }
-              lines.push(tarr);
-          }
-      }
```
Comment (Tony1023) [inline src/components/dashboard/menu-table/menu-table.js:228]: Is this done on the backend now?

File: src/components/dashboard/menu-table/popup-forms.js [hunk 5]
```diff
@@ -433,11 +435,140 @@ const QRCodeForm = (props) => {
...
+            reader.onabort = () => console.error('file reading was aborted')
+            reader.onerror = () => {
+                console.error('file reading has failed')
+                setErrorMessage("Error uploading file")
+            }
+            reader.onload = () => {
+            // Do whatever you want with the file contents
+              const fileContent = reader.result
+              setContent(fileContent)
+              setFileName(file.path)
```
Comment (Tony1023) [inline src/components/dashboard/menu-table/popup-forms.js:492]: I happened to drag in an empty file and the upload view did not respond to that. Nor did it prompt me about the empty file. Maybe we wanna take care of that? Like displaying file size maybe.

File: src/components/dashboard/menu-table/menu-table.js [hunk 6]
```diff
@@ -110,42 +124,23 @@ const MenuTable = (props) => {
     const [searchBoxValue, setSearchBoxValue] = useState(null);
     let fileReader
```
Comment (Tony1023) [inline src/components/dashboard/menu-table/menu-table.js:125]: Remove plz lol

File: src/components/dashboard/menu-table/menu-table.js (hunk 4 above)
Comment (cchyung) [reply src/components/dashboard/menu-table/menu-table.js:228]: yeet

File: src/components/dashboard/menu-table/popup-forms.js (hunk 1 above)
Comment (cchyung) [reply src/components/dashboard/menu-table/popup-forms.js:491]: Empty csv file or just generally empty file?

Comment (Tony1023) [reply src/components/dashboard/menu-table/popup-forms.js:481]: @cchyung An empty csv is what I tried. Not sure about an empty file of ANY format.

File: src/pages/dashboard/menu.js (hunk 2 above)
Comment (Tony1023) [reply src/pages/dashboard/menu.js:65]: Resolved. Looks like it's not styled.

**Pull request merged**